src/validators/apps/ecommerce/product.validators.js

```diff
@@ -20,7 +21,7 @@ const createProductValidator = () => {
       .withMessage("Stock is required")
       .isNumeric()
       .withMessage("Stock must be a number"),
-    body("category").notEmpty().isMongoId().withMessage("Invalid category id"),
+    ...mongoIdRequestBodyValidator("category")
```

Comment:
@wajeshubham

Here we have 2 choices. Either spread out our `mongoIdRequestBodyValidator` or else moving this to the route like below. Which one would you prefer ?

The reason I spread out is to keep all validators at one place. 


```
router
  .route("/")
  .get(getAllProducts)
  .post(
    verifyJWT,
    verifyPermission([UserRolesEnum.ADMIN]),
    // In product form we will received one main image file type
    // And max 4 sub images
    upload.fields([
      {
        name: "mainImage",
        maxCount: 1,
      },
      {
        // frontend will send at max 4 `subImages` keys with file object which we will save in the backend
        name: "subImages",
        maxCount: MAXIMUM_SUB_IMAGE_COUNT, // maximum number of subImages is 4
      },
    ]),
    createProductValidator(),
    mongoIdRequestBodyValidator("category"),
    validate,
    createProduct
  );

```

Reply:
Spread out the `mongoIdRequestBodyValidator` inside the validator itself, that will avoid updation of the routes which are close to the business logic.

Reply:
@wajeshubham Perfect then, The pushed version of code spreads out in validator itself.